pkg/audit/manager.go

```diff
@@ -79,6 +79,21 @@ type StatusViolation struct {
 	EnforcementAction string `json:"enforcementAction"`
 }
 
+// nsCache is used for caching namespaces and their labels
+type nsCache struct {
+	cache map[string]corev1.Namespace
+}
+
+func newNSCache() *nsCache {
+	return &nsCache{
+		cache: make(map[string]corev1.Namespace),
+	}
+}
+
+func (c *nsCache) getNSCache() map[string]corev1.Namespace {
```

Comment:
We would want this method to be `Get(namespace string)` so that we are wrapping the lazy lookup behind one method. This way nsCache can be passed around in the future without duplicating the check-cache-if-miss-call-client.get code.

Reply:
added